.github/workflows/build-test.yml

```diff
@@ -90,15 +90,22 @@ jobs:
       - run: ./tools/ci/build-test-ubuntu-with-sanitizers.sh gcc
 
   build-test-ubuntu-with-sanitizers-clang-latest:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-22.04
     needs: pre-flight-check
     steps:
-      - uses: egor-tensin/setup-clang@v1
-        with:
-          version: 13
-          platform: x64
-      - uses: actions/checkout@v3
-      - run: ./tools/ci/build-test-ubuntu-with-sanitizers.sh clang
+      - name: install newest clang version
+        run: |
+         wget https://apt.llvm.org/llvm.sh
+         chmod +x llvm.sh
+         sudo ./llvm.sh 15 all ## uses clang version 15
```

Comment:
Clang version 15 is the current development branch in llvm and still subject to changes. Do we really want to depend here on a version that may introduce bugs?
What if it introduces a bug that is removed again before making clang-15 stable? My recommendation would be to use clang-14 which is the latest stable version released by LLVM: https://releases.llvm.org/

The llvm.sh script sets the following: `CURRENT_LLVM_STABLE=14`.
We could evaluate this version number automatically before executing the script with
```
eval $(cat llvm.sh | grep CURRENT_LLVM_STABLE= -m 1)
```
and use that for installing the latest stable clang. With that we are saving effort on updating the clang-version every time.

Reply:
Ok regarding the Linting with clang-tidy it make sense to use clang-15 here. We can go with that for now but we should keep in mind that we need to update the clang versions at a later point when new clang releases were published.